ui/pages/confirm-approve/confirm-approve.js

```diff
@@ -199,6 +204,33 @@ export default function ConfirmApprove({
             toAddress={toAddress}
             tokenSymbol={tokenSymbol}
             decimals={decimals}
+            nextNonce={nextNonce}
+            customNonceValue={customNonceValue}
+            updateCustomNonce={(value) => {
+              dispatch(updateCustomNonce(value));
+            }}
+            getNextNonce={() => dispatch(getNextNonce())}
+            showCustomizeNonceModal={({
+              /* eslint-disable no-shadow */
+              useNonceField,
+              nextNonce,
+              customNonceValue,
+              updateCustomNonce,
+              getNextNonce,
+              /* eslint-disable no-shadow */
+            }) =>
+              dispatch(
+                showModal({
+                  name: 'CUSTOMIZE_NONCE',
+                  useNonceField,
+                  nextNonce,
+                  customNonceValue,
+                  updateCustomNonce,
+                  getNextNonce,
+                }),
+              )
+            }
+            warning={submitWarning}
```

Comment:
Function above is huge, a callback for this will look better.

Reply:
Also not sure why we need to pass doen. this function in props, it can be called in [token-allowance.js](https://github.com/MetaMask/metamask-extension/pull/17945/files#diff-86aefe0b412c9559b6b5ef7454c30a0c21fabf9e6303658be198ae85c7c13b52) directly.

Reply:
@jpuri Done [here](https://github.com/MetaMask/metamask-extension/pull/17945/commits/cb62dbcddb0b653bb306db0c89f7e9bc72cd4284).